reference.md

```diff
@@ -499,18 +542,154 @@ variables are missing:
 
 ``` fennel
 ;; bad
-(match [1 2 3]
+(case [1 2 3]
   ;; Will throw an error because `b' is nil for the first
   ;; pattern but the guard still uses it.
   (where (or [a 1 2] [a b 3]) (< a 0) (< b 1))
   :body)
 
 ;; ok
-(match [1 2 3]
+(case [1 2 3]
   (where (or [a b 2] [a b 3]) (< a 0) (<= b 1))
   :body)
 ```
 
+#### Binding Unification
+
+Symbols bound inside a `case` pattern are independent from any existing
+symbols in the current scope, that is - names may be re-used without
+consequence.
+
+Example:
+
+```fennel
+(let [x 1]
+  (case [:hello]
+    ;; `x` is simply bound to the first value of [:hello]
+    [x] x)) ; -> :hello
+```
+
+Sometimes it may be desirable to match against an existing value in the outer
+scope. To do this we can "unify" a binding inside the pattern with an existing
+outer binding with the unary `(= my-symbol)` form. The unary `(= my-symbol)`
+form is *only* valid in a `case` pattern.
+
+Example:
+
+```fennel
+(let [x 1]
+  (case [:hello]
+    ;; 1 != :hello
+    [(= x)] x
+    _ :no-match)) ; -> no-match
+
+(let [x 1]
+  (case [1]
+    ;; 1 == 1
+    [(= x)] x
+    _ :no-match)) ; -> 1
+
+(let [pass :hunter2]
+  (case (user-input)
+    (= pass) :login
+    _ :try-again!))
+```
+
+Unification is only required inside the pattern. Outer bindings are
+automatically available inside guards and bodies as long as the name has not
+been rebound in the pattern.
+
+Example:
+
+```fennel
+(let [limit 10]
+  (case [9 9]
+    ;; automatically avaliable in the guard
+    (where [a b] (< limit (+ a b))) :over-the-limit
+    ;; also available in the body
+    [a b] (+ a b limit))) ; -> :over-the-limit
+```
+
+If a pattern defines a binding with the same name as an existing binding in the
+outer scope, the inner binding takes precedence.
```

Comment:
Let's say instead:

    If a pattern binds a local with the same name as an existing local in the outer scope, the inner binding shadows the outer one.

The "shadows" terminology I think is a little more widespread for this. It might be helpful to reference forward here and mention that this is the difference between `case` and `match`; what do you think?